packages/xlucene-evaluator/src/parser/interfaces.ts

```diff
@@ -165,3 +177,23 @@ export interface Wildcard extends StringDataType, TermLikeAST {
 export interface Term extends AnyDataType, TermLikeAST {
     type: ASTType.Term;
 }
+
+export interface FunctionConfig {
+    logger: Logger;
+    typeConfig: TypeConfig;
+}
+
+export interface FunctionDefinition {
+    version: string;
+    name: string;
+    create: (
+        field: string,
+        params: any,
+        config: FunctionConfig
+        ) => FunctionMethods;
+}
+
+export interface FunctionMethods {
+    match(arg: any): boolean;
+    toElasticsearchQuery(options: UtilsTranslateQueryOptions): any;
```

Comment:
This should return `any` it should have a specific contract:
```suggestion
    toElasticsearchQuery(options: UtilsTranslateQueryOptions): { query: AnyQuery, sort?: AnySortQuery };
```

Reply:
fixed